### PR TITLE
Update tiny models and pipeline tests

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Update tiny models and pipeline tests.

The two failing pipeline tests for `rwkv` is addressed in #23442 and #23444 (and need to merge them before this one)

### CODE REVIEW

File: tests/models/gpt_neo/test_modeling_gpt_neo.py [hunk 1]
```diff
@@ -413,8 +413,8 @@ class GPTNeoModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMix
             "feature-extraction": GPTNeoModel,
             "question-answering": GPTNeoForQuestionAnswering,
             "text-classification": GPTNeoForSequenceClassification,
-            "token-classification": GPTNeoForTokenClassification,
             "text-generation": GPTNeoForCausalLM,
+            "token-classification": GPTNeoForTokenClassification,
```
Comment (ydshieh) [inline tests/models/gpt_neo/test_modeling_gpt_neo.py:417]: just the ordering thing

File: tests/models/gpt_neox/test_modeling_gpt_neox.py [hunk 2]
```diff
@@ -239,8 +239,8 @@ class GPTNeoXModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMi
             "feature-extraction": GPTNeoXModel,
             "question-answering": GPTNeoXForQuestionAnswering,
             "text-classification": GPTNeoXForSequenceClassification,
-            "token-classification": GPTNeoXForTokenClassification,
             "text-generation": GPTNeoXForCausalLM,
+            "token-classification": GPTNeoXForTokenClassification,
```
Comment (ydshieh) [inline tests/models/gpt_neox/test_modeling_gpt_neox.py:243]: just the ordering thing

File: tests/models/swiftformer/test_modeling_swiftformer.py [hunk 3]
```diff
@@ -139,7 +139,6 @@ class SwiftFormerModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestC
     """
 
     all_model_classes = (SwiftFormerModel, SwiftFormerForImageClassification) if is_torch_available() else ()
-
```
Comment (ydshieh) [inline tests/models/swiftformer/test_modeling_swiftformer.py:142]: just style

File: tests/test_pipeline_mixin.py [hunk 4]
```diff
@@ -263,6 +263,15 @@ def run_pipeline_test(self, task, repo_name, model_architecture, tokenizer_name,
...
 
+        pipeline_test_class_name = pipeline_test_mapping[task]["test"].__name__
+        if self.is_pipeline_test_to_skip_more(pipeline_test_class_name, model.config, model, tokenizer, processor):
+            logger.warning(
+                f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')} is skipped: test is "
+                f"currently known to fail for: model `{model_architecture.__name__}` | tokenizer "
+                f"`{tokenizer_name}` | processor `{processor_name}`."
+            )
+            return
+
```
Comment (ydshieh) [inline tests/test_pipeline_mixin.py:274]: See doc-string in the new `is_pipeline_test_to_skip_more` below

File: tests/utils/tiny_model_summary.json
Comment (ydshieh) [inline tests/utils/tiny_model_summary.json:1]: just update the information from some recently added models

**Review by amyeroberts — APPROVED** [review]
Nice! 

Thanks for adding and for taking the time to add PR comments - it makes the review super easy ❤️

**Pull request merged**